### ISSUE TITLE
Custom prelude

### ISSUE THREAD

**Issue by ModProg** [rune-rs/rune#538]
While it is possible to put custom functions/types in an unnamed module to achieve something similar, it is not possible to have the same type both in the root and a module.

I think all that would be needed for this would be to, make `Prelude`, `Prelude::with_default_prelude()` and `Prelude::add_prelude` public, and add a `with_prelude` function to `Build` or `Context`.

**Comment by udoprog** [comment]
Yeah, I want this too. Some kind of `ImplicitPrelude` object (to be named) would be nice which you can provide when building a `Context`.
